ui/js/utils.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Thin wrapper around localStorage.getItem(). Parses JSON and returns undefined if the value
+ * is not set yet.
+ */
+export function getLocal(key) {
+  const itemRaw = localStorage.getItem(key);
+  return itemRaw === null ? undefined : JSON.parse(itemRaw);
+}
+
+/**
+ * Thin wrapper around localStorage.setItem(). Converts value to JSON.
+ */
+export function setLocal(key, value) {
+  localStorage.setItem(key, JSON.stringify(value));
+}
+
+/**
+ * Records a publish attempt in local storage. Returns a dictionary with all the data needed to
+ * needed to make a dummy claim or file info object.
+ */
+export function savePendingPublish(name) {
+  const pendingPublishes = getLocal('pendingPublishes') || [];
+  const newPendingPublish = {
+    claim_id: 'pending_claim_' + name,
+    txid: 'pending_' + name,
+    nout: 0,
+    outpoint: 'pending_' + name + ':0',
+    name: name,
+    time: Date.now(),
+  };
+  setLocal('pendingPublishes', [...pendingPublishes, newPendingPublish]);
+  return newPendingPublish;
+}
+
+export function removePendingPublish({name, outpoint}) {
+  setLocal('pendingPublishes', getPendingPublishes().filter(
+    (pub) => pub.name != name && pub.outpoint != outpoint
+  ));
+}
+
+/**
+ * Gets the current list of pending publish attempts. Filters out any that have timed out and
+ * removes them from the list.
+ */
+export function getPendingPublishes() {
+  const pendingPublishes = getLocal('pendingPublishes') || [];
+
+  const newPendingPublishes = [];
+  for (let pendingPublish of pendingPublishes) {
+    if (Date.now() - pendingPublish.time <= lbry.pendingPublishTimeout) {
+      newPendingPublishes.push(pendingPublish);
+    }
+  }
+  setLocal('pendingPublishes', newPendingPublishes);
+  return newPendingPublishes
+}
+
+/**
+ * Gets a pending publish attempt by its name or (fake) outpoint. If none is found (or one is found
+ * but it has timed out), returns null.
+ */
+export function getPendingPublish({name, outpoint}) {
```

Comment:
I know it's awkward to use Local Storage for queryable, list-like data like this. The main alternatives would be IndexedDB (too complicated without a wrapper), Sqlite (possibly too complicated and would require doing stuff in the main process and IPC), or a light JS key/value database like nedb (adds a dependency and stuff to learn).

If we don't like local storage, it would be fairly quick to switch to something like nedb.